### DISCUSSION TITLE
Update to new version

### DISCUSSION THREAD

**Discussion by OpenSourceSimon** [plausible/analytics#3691] [Anything Else]
Hi all, when's the new update of Plausible for self-hosted systems? I saw some great improvements in the changelog, and they work perfectly on the Cloud version! It has been over 6 months since the latest update (V2.0) came out on Docker.

**Comment by Cretezy** [comment]
I have setup a CI to build from the master branch and deploy to GitHub Container Registry here: https://github.com/Cretezy/Plausible (the image is `ghcr.io/cretezy/plausible`).

Newer versions of Plausible require `TOTP_VAULT_KEY` to be set (in your `plausible-conf.env`). You can generate one using `openssl rand -base64 32 | tr -d '\n' ; echo`

Still a WIP, would like to add so it pulls directly from `plausible/analytics` and runs when a commit/tag is pushed, but for now it's on latest.

Use at your own risk (there's no upgrade information available).

**Reply by raabmar** [reply]
@Cretezy this is some unofficial update then right? I only tried once to update with some commands for docker

WIP means work in progress?

**Reply by Cretezy** [reply]
It is an unofficial distribution of the official code (from the `master` branch). 

If you don't know how to use Docker, this is likely not for you, since this may break your installation of Plausible.

**Reply by raabmar** [reply]
Thanks - unfortunately I can't do that. It was already difficult to get it up and running with a good guide line. I have a page saved with instructions on how to update it. It's only 2 or 3 commands in the command line. However, that must be the official way then. I am surprised that there is hardly any official update available at all. My reason to go with Plausible was to get it self hosted (I am not charging any client for this so SaaS would become expensive soon).

Too bad, updates are not happening.

**Reply by PetervdPerk** [reply]
@raabmar You're using the following docker compose file? https://github.com/plausible/hosting/blob/master/docker-compose.yml
I've modified it to use your `ghcr.io/cretezy/plausible:latest` instead of `plausible/analytics:v2.0`
But I'm getting errors trying to access port 8000 
```
peter@desktop-peter:~/hosting$ wget localhost:8000
--2024-02-12 22:56:28--  http://localhost:8000/
Resolving localhost (localhost)... 127.0.0.1
Connecting to localhost (localhost)|127.0.0.1|:8000... connected.
HTTP request sent, awaiting response... Read error (Connection reset by peer) in headers.
Retrying.
```

Even removed all docker volumes and did a clean install but http server doesn't work for me.
If you've got tag of a working that would be much appreciated.

**Reply by raabmar** [reply]
@PetervdPerk I think you tagged the wrong person. @Cretezy is probably needed here

**Reply by Cretezy** [reply]
Newer versions of Plausible require `TOTP_VAULT_KEY` to be set (in your `plausible-conf.env`). You can generate one using `openssl rand -base64 32 | tr -d '\n' ; echo`

(Added this information to the post above. Please check/include logs next time when posting an error)

**Comment by OpenSourceSimon** [comment]
After doing some research I've found that Plausible uses an external DockerHub repository to publish their latest changes on the master branch. You can just pull from `plausible/analytics-private:latest` to get the latest code. 

However, it looks like this is the code of the cloud version. I'm on a 'trial' for 35632 days now 😀. Also there's a whole page on subscriptions and it looks like I can even pay to upgrade my free long 'trial'. This just looks like a bug because it's built for the cloud version, I can still use all the (new) features. 

Check-out the Docker Hub website for more information: https://hub.docker.com/r/plausible/analytics-private/tags

**Reply by raabmar** [reply]
Overall, it looks there are a lot of reasons to reconsider the self-hosted version. I thought with Docker once you got through the setup and got it up and running it means simply running one or two commands to update. 

Are all these discussions and manual steps required because Plausible doesn't officially get updated (and perhaps the official update path happens only once per year to get more clients onto the cloud version?)

I am not charging my clients for Plausible, because they are used to free GA4, and it's already difficult enough to get them away from it. However, even though I am not charging them, I would like to be sure to be able to run updates of Plausible - and till now that seems to be overly complicated. What do you think about that?

I perhaps could still switch to another solution or just keep Plausible running as is (it won't fall apart when not updating for some time)

**Reply by zell-mbc** [reply]
Once you switched over to :latest you are pretty much done.
All that's required is a `sudo docker-compose down --rmi all || sudo docker-compose up -d ` every now and then.

The only issue with that is that you will be getting whatever 'latest' at this point in time is, this may be fine if latest is of good quality, but could be problematic if latest is beta quality.
I haven't decided yet if I switch over to latest / master…still pretty new to Plausible.

<it looks like this is the code of the cloud version.>
@OpenSourceSimon I am running the official v2 and got emails saying that I'm in my trial period as well. Probably means that the code doesn't distinguish between cloud or self-hosted, no matter which image you pull.

By the way, the docker-compose file says only Postgres 12-14 is supported. Well, I started withPostgres 16, so far no issues…

**Reply by ruslandoga** [reply]
> By the way, the docker-compose file says only Postgres 12-14 is supported. Well, I started withPostgres 16, so far no issues…

Right, we'll need to update our CI to test against more recent PostgreSQL versions and change that comment. Thank you!

**Comment by ruslandoga** [comment]
[First release candidate for v2.1.0](https://github.com/plausible/analytics/discussions/3817) is now available!

**Reply by ruslandoga** [reply]
And now the second one is available too: https://github.com/plausible/analytics/discussions/4084
